cli/util/file_watcher.rs

```diff
@@ -304,6 +304,9 @@ where
   }
 
   loop {
+    // We may need to give the runtime a tick to settle, as cancellations may need to propagate to tasks
+    tokio::time::sleep(Duration::from_millis(1)).await;
```

Comment:
I'm not really happy with this part of the PR -- but it's very difficult to get at the resource table from the watcher. Ideally we should be shutting down all resources in the resource table _if_ we're in watch mode, and letting things drop if we're in a normal run mode.

Reply:
What is the problem with dropping `OpState` and `ResourceTable` in the watcher - if it's done properly I'd expect everything to be closed done cleanly. Then we can `yield()` 10 times after the drop to make sure that Tokio has enough ticks to resolve everything.

Reply:
I think to be really "correct" we should be explicitly `close`ing all the resources in the resource table on watch restart to ensure that all cancellations are triggered and propagate correctly to futures, including those inside of tasks. As it stands right now, we don't really have a good way to 1) start the close process for every resource and 2) confirm that the close process for every resource has completed.